mycroft/skills/core.py

```diff
@@ -699,9 +711,45 @@ def cancel_event(self, name):
             Args:
                 name (str):   Name of event
         """
-        data = {'event': self._unique_name(name)}
+        unique_name = self._unique_name(name)
+        data = {'event': unique_name}
+        self.remove_event(unique_name)
         self.emitter.emit(Message('mycroft.scheduler.remove_event', data=data))
 
+    def get_event_status(self, name):
+        """
+            Get event data and return the amount of time left
+
+            Args:
+                name (str): Name of event
+
+            Return:
+                time_left (int): the time left in seconds
+        """
+        event_name = self._unique_name(name)
+        data = {'name': event_name}
+
+        # making event_status an object so it's refrence can be changed
+        event_status = [None]
+
+        def callback(message, event_status):
```

Comment:
event_status don't seem to be needed as an argument

Reply:
this is actually needed to be able to change the value of the event_status. If this is not passed this way to the callback then when we say event_status = whatever, it won't reference the same variable.

Reply:
Hmm, when I tested it wasn't needed. As long the outer_var is a mutable (in this case a list) it should work.

```python
def outer():
    outer_var = [None]
    def inner():
        outer_var[0] = 'Modified'

    inner()
    print outer_var[0]

outer()
```

I'll retest but it's not really important, this way works as well.

Reply:
Ok retried and it doesn't seem necessary but never mind both ways work.

Reply:
ahh ok thanks!